src/ccache/ccache.cpp

```diff
@@ -2492,10 +2492,6 @@ do_cache_compilation(Context& ctx)
 
   ProcessArgsResult processed = process_args(ctx);
 
-  if (processed.error) {
-    return tl::unexpected(*processed.error);
-  }
-
```

Comment:
This condition can never be false, it's useless? If you look [here](https://github.com/ccache/ccache/blob/master/src/ccache/argprocessing.cpp#L1747-L1752) and [here](https://github.com/ccache/ccache/blob/master/src/ccache/argprocessing.hpp#L34-L41).

Reply:
You seem to have missed the https://github.com/ccache/ccache/blob/baea34b407f5f4f036fce8ce6f5d7873c572467f/src/ccache/argprocessing.hpp#L33 constructor, for instance invoked here: https://github.com/ccache/ccache/blob/baea34b407f5f4f036fce8ce6f5d7873c572467f/src/ccache/argprocessing.cpp#L1286

Reply:
The construction comes from before `tl::expected` was introduced. I'll refactor it to use `tl::expected` instead.

Reply:
I see there are 13 of them 😆 I totally missed it, the method is so long.

Reply:
I have reverted this change.

Reply:
> I'll refactor it to use tl::expected instead.

Done in 181ffcff3186.